### PR TITLE
Update template for adding basic auth and guidance

### PR DESCRIPTION
**Pull request by kateriley1**
## Pull request checklist

- [x] Add a descriptive message for this change to the PR
- [ ] Run `black ./` locally
- [ ] Run `pylint <dashboard name>` locally and obtain a score of 10.00/10.00
- [ ] Run `python -u -m pytest --headless tests` locally and there are no failures
- [ ] Include screenshot for any visual changes
- [ ] Use cds_to_staging.py to push any new or updated data to the S3 bucket

### PR Description:
- update package version
- update readme guidance for adding basic auth
- remove bootsrap.py as setup_application_http_response_headers function is in package

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -76,10 +76,22 @@ cf map-route <APP_NAME> <DOMAIN> --hostname <HOSTNAME>
...
-    You will need to have access to the `cf` command installed and configured, which currently isn't the case within the DAP.
-    GOV.UK PaaS provide guidance on how to do this under the title [Example: Route service to add username and password authentication][basic_auth].
+
+## OPTIONAL: Create a shared username/password for accessing the hosted dashboard
+This can be useful if you want to prevent curious individuals from accessing your dashboard while in development, but does not give any security against malicious actors.
+You will need to have access to the `cf` command installed and configured, which can be requested through DAP support.
+
+GOV.UK PaaS provide guidance on how to do this under the title [Example: Route service to add username and password authentication][basic_auth], however following this method you cannot have both basic authentication and IP restricitons in place as your app can only have one routing service. If you would like your app to have basic authentication and IP restricitons, please follow the below steps to add basic authentication using flask:
+
+1. In `bootstrap.py` uncomment the `enable_basic_auth` function
```
Comment (adrianclay) [inline README.md:86]: Won't the `enable_basic_auth` method need to be called by something?  I'm not confident uncommenting by itself would be good enough?

I'm wondering if it makes sense to add this method to our shared library if it's going to be shared across multiple dashboards?

**Review by tom-clarke-dluhc — APPROVED** [review]

**Review by duncanc19 — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -76,10 +76,26 @@ cf map-route <APP_NAME> <DOMAIN> --hostname <HOSTNAME>
...
+This can be useful if you want to prevent curious individuals from accessing your dashboard while in development, but does not give any security against malicious actors.
+
+GOV.UK PaaS provide guidance on how to do this under the title [Example: Route service to add username and password authentication][basic_auth], you will need to have access to the `cf` command installed and configured, which can be requested through DAP support. However following the GOV.UK PaaS method you cannot have both basic authentication and IP restricitons in place as your app can only have one routing service. If you would like your app to have basic authentication and IP restrictions, please follow the below steps to add basic authentication using flask:
+
+1. In your GitHub repository, add environment secrets for `APP_USERNAME` and `APP_PASSWORD`
+1. In `deployment.yml` under the `deploy-staging` job, for the task `Set environment variables` add the below to the run command:
+```python 
+cf set-env ${{ env.application-name }} APP_USERNAME ${{ secrets.APP_USERNAME }}
+cf set-env ${{ env.application-name }} APP_PASSWORD ${{ secrets.APP_PASSWORD }}
+```
```
Comment (adrianclay) [inline README.md:90]: Currently the numbering is interrupted and restarts at 1.

If you indent the codeblock here, then GitHub should treat the app.py instruction as part of the steps above.

```suggestion
   ```python 
   cf set-env ${{ env.application-name }} APP_USERNAME ${{ secrets.APP_USERNAME }}
   cf set-env ${{ env.application-name }} APP_PASSWORD ${{ secrets.APP_PASSWORD }}
   ```
```

Comment (adrianclay) [reply README.md:90]: Hmm, strange that that suggested change removes the ``` even though I included it.

Comment (kateriley1) [reply README.md]: Good spot - I've committed this change ending with ```

**Pull request merged**